pkg/windows/calico.go

```diff
@@ -285,106 +358,55 @@ func checkForCorrectInterface() (bool, error) {
-	if _, err := hcsshim.GetHNSNetworkByName(n); err != nil {
-		return false
-	}
-	return true
-}
-
-func createExternalNetwork(backend string, vxlanAdapter string) error {
-	logrus.Debug("Creating external network")
-	for !(checkIfNetworkExists(CalicoHnsNetworkName)) {
-		logrus.Debugf("Networking doesn't exist yet: %s ", CalicoHnsNetworkName)
-		var network hcsshim.HNSNetwork
-		if backend == "vxlan" {
-			logrus.Debugf("Backend is vxlan")
-			// Ignoring the return because both true and false without an error represent that the firewall rule
-			// was created or already exists
-			if _, err := wapi.FirewallRuleAdd(
-				"OverlayTraffic4789UDP",
-				"Overlay network traffic UDP",
-				"",
-				"4789",
-				wapi.NET_FW_IP_PROTOCOL_UDP,
-				wapi.NET_FW_PROFILE2_ALL,
-			); err != nil {
-				logrus.Debugf("error creating firewall rules: %s", err)
-				return err
-			}
-			logrus.Debugf("Creating VXLAN network using the vxlanAdapter: %s", vxlanAdapter)
-			network = hcsshim.HNSNetwork{
-				Type:               "Overlay",
-				Name:               CalicoHnsNetworkName,
-				NetworkAdapterName: vxlanAdapter,
-				Subnets: []hcsshim.Subnet{
-					{
-						AddressPrefix:  "192.168.255.0/30",
-						GatewayAddress: "192.168.255.1",
-						Policies: []json.RawMessage{
-							[]byte("{ \"Type\": \"VSID\", \"VSID\": 9999 }"),
-						},
+		logrus.Debugf("Creating VXLAN network using the vxlanAdapter: %s", vxlanAdapter)
+		network = hcsshim.HNSNetwork{
+			Type:               "Overlay",
+			Name:               CalicoHnsNetworkName,
+			NetworkAdapterName: vxlanAdapter,
+			Subnets: []hcsshim.Subnet{
+				{
+					AddressPrefix:  "192.168.255.0/30",
+					GatewayAddress: "192.168.255.1",
+					Policies: []json.RawMessage{
+						[]byte("{ \"Type\": \"VSID\", \"VSID\": 9999 }"),
 					},
 				},
-			}
-		} else {
-			network = hcsshim.HNSNetwork{
-				Type: "L2Bridge",
-				Name: CalicoHnsNetworkName,
-				Subnets: []hcsshim.Subnet{
-					{
-						AddressPrefix:  "192.168.255.0/30",
-						GatewayAddress: "192.168.255.1",
-					},
+			},
+		}
+	} else {
+		network = hcsshim.HNSNetwork{
+			Type: "L2Bridge",
+			Name: CalicoHnsNetworkName,
+			Subnets: []hcsshim.Subnet{
+				{
+					AddressPrefix:  "192.168.255.0/30",
+					GatewayAddress: "192.168.255.1",
 				},
-			}
+			},
 		}
+	}
+
+	if _, err := network.Create(); err != nil {
+		return "", fmt.Errorf("Error creating the %s network: %v", CalicoHnsNetworkName, err)
+	}
 
-		logrus.Debugf("Creating network.")
-		if _, err := network.Create(); err != nil {
-			logrus.Debugf("waiting for network %s", err)
-			time.Sleep(1 * time.Second)
+	// Check if network exists. If it does not after 5 minutes, fail
+	for start := time.Now(); time.Since(start) < 5*time.Minute; {
+		network, err := hcsshim.GetHNSNetworkByName(CalicoHnsNetworkName)
+		if err == nil {
+			return network.ManagementIP, nil
 		}
 	}
-	return nil
+
+	return "", fmt.Errorf("Failed to create %s network", CalicoHnsNetworkName)
```

Comment:
Error strings shouldn't contain punctuation.

```suggestion
	return "", fmt.Errorf("failed to create %s network", CalicoHnsNetworkName)
```

Reply:
good to know